internal/interfaces/storage_volume_efficiency_policies.go

```diff
@@ -0,0 +1,185 @@
+func GetStorageVolumeEfficiencyPoliciesByUUID(errorHandler *utils.ErrorHandler, r restclient.RestClient, uuid string) (*StorageVolumeEfficiencyPoliciesGetDataModelONTAP, error) {
+	api := "storage/volume-efficiency-policies/" + uuid
+	query := r.NewQuery()
+	query.Fields([]string{"name", "svm.name", "type", "qos_policy", "comment", "enabled", "schedule", "duration", "start_threshold_percent", "uuid"})
+	statusCode, response, err := r.GetNilOrOneRecord(api, query, nil)
+	if err == nil && response == nil {
+		err = fmt.Errorf("no response for GET %s", api)
+	}
+	if err != nil {
+		return nil, errorHandler.MakeAndReportError("error reading storage_volume_efficiency_policies info", fmt.Sprintf("error on GET %s: %s, statusCode %d", api, err, statusCode))
+	}
+
+	var dataONTAP StorageVolumeEfficiencyPoliciesGetDataModelONTAP
+	if err := mapstructure.Decode(response, &dataONTAP); err != nil {
+		return nil, errorHandler.MakeAndReportError(fmt.Sprintf("failed to decode response from GET %s", api),
+			fmt.Sprintf("error: %s, statusCode %d, response %#v", err, statusCode, response))
+	}
+	tflog.Debug(errorHandler.Ctx, fmt.Sprintf("Read storage_lun data source: %#v", dataONTAP))
+	return &dataONTAP, nil
+}
+
+// GetStorageVolumeEfficiencyPoliciesByName to get storage_volume_efficiency_policies info
+func GetStorageVolumeEfficiencyPoliciesByName(errorHandler *utils.ErrorHandler, r restclient.RestClient, name string, svmName string) (*StorageVolumeEfficiencyPoliciesGetDataModelONTAP, error) {
+	api := "storage/volume-efficiency-policies"
+	query := r.NewQuery()
+	query.Set("name", name)
+	query.Set("svm.name", svmName)
+	query.Fields([]string{"name", "svm.name", "type", "qos_policy", "comment", "enabled", "schedule", "duration", "start_threshold_percent", "uuid"})
+	statusCode, response, err := r.GetNilOrOneRecord(api, query, nil)
+	if err == nil && response == nil {
+		err = fmt.Errorf("no response for GET %s", api)
+	}
+	if err != nil {
+		return nil, errorHandler.MakeAndReportError("error reading storage_volume_efficiency_policies info", fmt.Sprintf("error on GET %s: %s, statusCode %d", api, err, statusCode))
+	}
+
+	var dataONTAP StorageVolumeEfficiencyPoliciesGetDataModelONTAP
+	if err := mapstructure.Decode(response, &dataONTAP); err != nil {
+		return nil, errorHandler.MakeAndReportError(fmt.Sprintf("failed to decode response from GET %s", api),
+			fmt.Sprintf("error: %s, statusCode %d, response %#v", err, statusCode, response))
+	}
+	tflog.Debug(errorHandler.Ctx, fmt.Sprintf("Read storage_volume_efficiency_policies data source: %#v", dataONTAP))
+	return &dataONTAP, nil
+}
+
+// GetStorageVolumeEfficiencyPoliciess to get storage_volume_efficiency_policies info for all resources matching a filter
+func GetStorageVolumeEfficiencyPoliciess(errorHandler *utils.ErrorHandler, r restclient.RestClient, filter *StorageVolumeEfficiencyPoliciesDataSourceFilterModel) ([]StorageVolumeEfficiencyPoliciesGetDataModelONTAP, error) {
```

Comment:
should it be "policy" and "policies"? I think the only place we should use double 's' is dns.